app/furniture/marketplace/carts/_cart.html.erb

```diff
@@ -1,20 +1,23 @@
 <div id="<%= dom_id(cart) %>">
 
-  <%- if cart.delivery_address.blank? %>
+  <%- if cart.delivery_address.blank? || cart.contact_phone_number.blank? %>
     <%= form_with model: cart.location do |cart_form| %>
       <%= render "text_field", attribute: :delivery_address, form: cart_form %>
+      <%= render "text_field", attribute: :contact_phone_number, form: cart_form %>
 
-      <%= cart_form.submit t('marketplace.delivery_address.edit') %>
+      <%= cart_form.submit t('marketplace.edit_delivery_info.edit') %>
     <%- end %>
   <%- else %>
     <div class="flex justify-between flex-wrap">
       <div class="w-full flex items-center justify-between text-sm">
         <span>Delivering to:</span>
+        <%= cart.delivery_address %>
         <%= render ButtonComponent.new(href: cart.location.concat([params: { cart: { delivery_address: nil } }]), label: t('marketplace.delivery_address.edit'), title: nil, classes: "shrink font-light text-xs m-0 bg-primary-200 underline") %>
-      </div>
-
 
-      <p class="grow"><%= cart.delivery_address %></p>
```

Comment:
If this was a `div` that was a sibling of the `div` that contains the "Delivering To" text, and had the same classes it would likely stack on smaller screens. 

We may need to add `sm:w-1/2` or `sm:w-auto` to make it take up half the space on larger screens

Reply:
Nice!